File: problems/HackerRank/breakingRecords.ts
function breakingRecords(scores: number[]): number[] {
  let max = scores[0];
  let min = scores[0];
  let maxCount = 0;
  let minCount = 0;

  for (let i = 1; i < scores.length; i++) {
    const score = scores[i];
    if (score < min) {
      min = score;
      minCount++;
    }

    if (score > max) {
      max = score;
      maxCount++;
    }
  }

  return [maxCount, minCount];
}

console.log(breakingRecords([10, 5, 20, 20, 4, 5, 2, 25, 1]));

File: problems/HackerRank/week1/breakingRecords.py
def breakingRecords(scores):
    min_score = max_score = scores[0]
    min_count = max_count = 0
    
    for score in scores[1:]:
        
        if score > max_score:
            max_count += 1
            max_score = score
        
        elif score < min_score:
            min_count += 1
            min_score = score
        
    return [max_count, min_count]  